src/iuse.cpp

```diff
@@ -4904,47 +4902,48 @@ static bool heat_item( player &p )
     }
     item &target = heat->is_food_container() ? heat->contents.front() : *heat;
     p.mod_moves( -300 ); //initial preparations
+    // simulates heat capacity of food, more weight = longer heating time
+    // this is x2 to simulate larger delta temperature of frozen food in relation to
+    // heating non-frozen food (x1); no real life physics here, only aproximations
+    int move_mod = ( to_gram( target.weight() ) );
+    // links time of food's HOT-ness with weight, as smaller items lose temperature faster
+    // locked in brackets between 10 minutes min and 60 minutes max to cut-off extreme values
+    int counter_mod = clamp( to_gram( target.weight() ), 100, 600 );
     if( target.item_tags.count( "FROZEN" ) ) {
-        add_msg( _( "You defrost the food." ) );
         target.item_tags.erase( "FROZEN" );
-        if( target.has_flag( "EATEN_COLD" ) ) {
-            // heat just enough to thaw it
-            p.mod_moves( -to_gram( target.weight() ) );
+        if( target.has_flag( "FREEZERBURN" ) ) {
+            if( !target.item_tags.count( "MUSHY" ) ) {
+                target.item_tags.insert( "MUSHY" );
+            } else {
+                target.set_relative_rot( 1.1 );
+            }
+        }
+
+        if( target.has_flag( "EATEN_COLD" ) &&
+            !query_yn( _( "%s is best served cold. Defrost and heat?" ), target.tname() ) ) {
```

Comment:
May be misunderstood along the lines of "of course I want to heat it in order to defrost it".
"Heat beyond defrosting?" might convey the choice better.

Reply:
Yeah I had a hard time trying to come up with something here, I like your suggestion, will fix